pandas/tests/frame/methods/test_to_arrays.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+from numpy import array
+
+import pandas._testing as tm
+from pandas.core.indexes.api import ensure_index
+from pandas.core.internals.construction import to_arrays
+
+
+def test_to_arrays():
+    # GH 59717
+    data = np.array(
+        [
+            ("John", 25, "New York", 50000),
+            ("Jane", 30, "San Francisco", 75000),
+            ("Bob", 35, "Chicago", 65000),
+            ("Alice", 28, "Los Angeles", 60000),
+        ],
+        dtype=[("name", "U10"), ("age", "i4"), ("city", "U15"), ("salary", "i4")],
+    )
+
+    columns = ["name", "salary", "city"]
+    indexed_columns = ensure_index(columns)
+
+    actual_arrays, actual_cols = to_arrays(data, indexed_columns)
```

Comment:
Could you tests `DataFrame.from_records` specifically and put the test in `pandas/tests/frame/constructors/test_from_records.py`?